### PR TITLE
Dev/rebuild plugin system

### PR DESCRIPTION
**Pull request by binyamin555**

### CODE REVIEW

File: core/ez/__init__.py [hunk 1]
```diff
@@ -42,30 +42,34 @@
 from utilities.event import Event
 from utilities.event_emitter import EventEmitter
 from web.response import _EzResponse
-from plugins.manager import PluginManager
-from plugins.installer import PluginInstaller
-from plugins.errors import UnknownPluginError
-from plugins.plugin_info import PluginInfo
+# from plugins.manager import PluginManager
+# from plugins.installer import PluginInstaller
```
Comment (neriyaco) [inline core/ez/__init__.py]: Is there a reason for the comments? Can we remove it completely?

File: core/ez/__init__.py [hunk 2]
```diff
@@ -42,30 +42,34 @@
...
 class _EZ:
     ez: "_EZ | None" = None
     ee: EventEmitter
     app: FastAPI
 
     plugin_events: dict[str, list[tuple[str, Callable]]]
-    plugin_manager: PluginManager
-    plugin_installer: PluginInstaller
+    # plugin_manager: PluginManager
+    # plugin_installer: PluginInstaller
```
Comment (neriyaco) [inline core/ez/__init__.py]: Comments here as well

File: core/ez/__init__.py [hunk 3]
```diff
@@ -78,8 +82,8 @@ def __init__(self, app: FastAPI | None = None, ee: EventEmitter | None = None) -
         self.app = app or EZApplication(redirect_slashes=True)
 
         self.plugin_events = {}
-        self.plugin_manager = PluginManager(PLUGINS_DIR)
-        self.plugin_installer = PluginInstaller(self.plugin_manager)
+        # self.plugin_manager = PluginManager(PLUGINS_DIR)
+        # self.plugin_installer = PluginInstaller(self.plugin_manager)
```
Comment (neriyaco) [inline core/ez/__init__.py]: Same comment

File: core/ez/__init__.py [hunk 4]
```diff
@@ -95,41 +99,55 @@ def remove_plugin_events(self, plugin: str):
...
 
 
-def get_plugin_event_handler_info(f) -> PluginInfo:
-    if not is_plugin_event_handler(f):
-        raise ValueError(f"Function '{f.__qualname__}' is not a plugin event handler.")
-    return getattr(f, "__ez_plugin__")
+# def get_plugin_event_handler_info(f) -> PluginInfo:
+#     if not is_plugin_event_handler(f):
+#         raise ValueError(f"Function '{f.__qualname__}' is not a plugin event handler.")
+#     return getattr(f, "__ez_plugin__")
```
Comment (neriyaco) [inline core/ez/__init__.py]: Comments

File: core/ez/__init__.py [hunk 5]
```diff
@@ -400,13 +418,17 @@ def _setup(__ez=_EZ.ez):
...
-    emit(Plugins.WillLoad)
-    load_plugins()
-    emit(Plugins.DidLoad)
+    emit(PluginEvent.WillLoad)
+    # TODO: WTF????
+    __pm.load_plugins(
+        "test-plugin",
+        "title-changer"
+    )
+    emit(PluginEvent.DidLoad)
```
Comment (neriyaco) [inline core/ez/__init__.py:372]: `iterdir`?

File: modules/plugins/ez_plugins/machinery.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+from ....machinery.installer import IPluginInstaller, PluginInstallerInfo, PluginInstallationResult
+from ....machinery.loader import IPluginLoader, PluginLoaderInfo
+from ....plugin_info import PluginInfo
```
Comment (neriyaco) [inline modules/plugins/ez_plugins/machinery.py:3]: What's with all this relative imports? Does it have to be relative?

**Review by neriyaco — COMMENTED** [review]
Nice.
Just remove comments if they are not needed anymore.

File: modules/plugins/ez_plugins/machinery.py (hunk 6 above)
Comment (binyamin555) [reply modules/plugins/ez_plugins/machinery.py:3]: Fixed

File: core/ez/__init__.py (hunk 5 above)
Comment (binyamin555) [reply core/ez/__init__.py:372]: No, it should pull the data from the database

Comment (neriyaco) [reply core/ez/__init__.py:372]: Ok makes sense

**Review by neriyaco — APPROVED** [review]

**Pull request merged**